libsql-server/src/auth/user_auth_strategies/http_basic.rs

```diff
@@ -7,27 +7,31 @@ pub struct HttpBasic {
 }
 
 impl UserAuthStrategy for HttpBasic {
-    fn authenticate(
-        &self,
-        context: Result<UserAuthContext, AuthError>,
-    ) -> Result<Authenticated, AuthError> {
+    fn authenticate(&self, ctx: UserAuthContext) -> Result<Authenticated, AuthError> {
         tracing::trace!("executing http basic auth");
+        let auth_str = ctx
+            .custom_fields
+            .get("authorization")
```

Comment:
should we use the constant from the `http` crate here instead of hard coding this value? If the hashmap needs all lower case we could wrap the hashmap to always call `to_lower_case` to ensurse that we always hash the correct key.

We use this hard coded value in a lot of places it would be nice if we could dry that up. I feel somewhat uneasy about having a dynamic hashmap that has hard coded values in diff spots all over the code base, I think we should make this a bit more type safe since this code is pretty critical and called an almost every request.